ksql-benchmark/src/main/java/io/confluent/ksql/benchmark/SerdeBenchmark.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.benchmark;
+
+import static io.confluent.ksql.datagen.DataGenSchemaUtil.getOptionalSchema;
+
+import io.confluent.avro.random.generator.Generator;
+import io.confluent.connect.avro.AvroData;
+import io.confluent.kafka.schemaregistry.client.MockSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.datagen.RowGenerator;
+import io.confluent.ksql.datagen.SessionManager;
+import io.confluent.ksql.serde.avro.KsqlAvroTopicSerDe;
+import io.confluent.ksql.serde.json.KsqlJsonTopicSerDe;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.Pair;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.Schema;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+// CHECKSTYLE_RULES.OFF: HideUtilityClassConstructor ; Reason: "@State annotation can only be
+// applied to classes having the default public constructor."
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Warmup(iterations = 5, time = 10)
+@Measurement(iterations = 5, time = 10)
+@Threads(4)
```

Comment:
If we want this to be run as part of build it'd be nice to decrease the overall time. We should figure out hte variance and try to play around with:
- Number of forks
- Number of warmups
- Any state that can be generated once per Benchmark instead of each iteration
- Total time per iteration

Perhaps useful data: I ran it with 1 fork and got
```
Benchmark                   (schemaName)  (serializationFormat)  Mode  Cnt  Score   Error  Units
SerdeBenchmark.deserialize   impressions                   JSON  avgt    5  1.331 ± 0.182  us/op
SerdeBenchmark.deserialize   impressions                   Avro  avgt    5  2.102 ± 0.053  us/op
SerdeBenchmark.deserialize       metrics                   JSON  avgt    5  6.891 ± 0.145  us/op
SerdeBenchmark.deserialize       metrics                   Avro  avgt    5  7.206 ± 0.166  us/op
SerdeBenchmark.serialize     impressions                   JSON  avgt    5  0.718 ± 0.016  us/op
SerdeBenchmark.serialize     impressions                   Avro  avgt    5  1.218 ± 0.024  us/op
SerdeBenchmark.serialize         metrics                   JSON  avgt    5  2.875 ± 0.080  us/op
SerdeBenchmark.serialize         metrics                   Avro  avgt    5  4.221 ± 0.201  us/op
```

Reply:
Agreed, will do in a follow-up PR. Thanks for the data -- I assume you got these results running on your laptop?

Reply:
Yup - just on my laptop